### PR TITLE
Allow configuration of ASG termination policies

### PR DESCRIPTION
**Pull request by atayarani**
@Tapjoy/eng-group-ops @StabbyCutyou 

This allows more flexibility around which instances get scaled out.

### CODE REVIEW

**Comment by StabbyCutyou** [comment]
Makes sense, but failing specs

**Comment by atayarani** [comment]
@StabbyCutyou specs are fixed

**Comment by atayarani** [comment]
@StabbyCutyou @marcuswalser @ehealy 
/cc @Tapjoy/eng-group-ops 

Accidentally made this PR bigger, but WONTFIX.

New notes:
* Laid the groundwork for static and dynamic scaling options
* Cleaned up a lot of stuff
* Deprecated/removed --config-dir.  It was confusing, but is no longer.

File: bin/tass [hunk 1]
```diff
@@ -78,17 +82,27 @@ when 'audit'
...
+    common_args
   end
 
-  config, aws_env, user_data = Tapjoy::AutoscalingBootstrap::Base.new.configure_environment(opts[:filename], opts[:env], opts[:config_dir])
-  Aws.config[:region] = config[:aws_region]
+  config, aws_env, user_data = configure_environment(opts)
 
   config.merge!(aws_env.merge(user_data: Base64.encode64("#{user_data}")))
   Tapjoy::AutoscalingBootstrap::Audit.new(config)
+when 'scale'
```
Comment (marcuswalser) [inline bin/tass]: We'd discussed potentially having min/max stored in the yaml file. Desired probably ought to remain a CLI option - maybe include the ability to do it either way?

**Comment by marcuswalser** [comment]
This is LGTM, we can address the min/max stuff in a followup PR.

**Pull request merged**